java/monkey/src/main/java/root/parser/Parser.java

```diff
@@ -0,0 +1,112 @@
+package root.parser;
+
+import root.Token;
+import root.TokenType;
+import root.ast.*;
+import root.ast.expressions.IdentiferExpression;
+import root.ast.statements.LetStatement;
+import root.ast.statements.ReturnStatement;
+import root.ast.statements.Statement;
+import root.lexer.Lexer;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class Parser {
```

Comment:
I think we could probably do something here with the `LexerIterable`, but I'm too tired this morning to come up with smart ideas.

Reply:
If you think about it, the Lexer itself already is an iterator (you call nextToken() until it returns EOF), the only advantage of declaring a custom Iterable is being able to use java default functions, so I don't think we would gain much from this.

Reply:
This would be more like an `Iterable<Statement>`, but I'm thinking the right design will become more clear as we progress forward.